### ISSUE TITLE
folder_contents navigation action fires double set of ajax requests

### ISSUE THREAD

**Issue by fredvd** [plone/Products.CMFPlone#3203]
Plone 5.2.X .  While working on the issue with the toolbar not being updated while navigation in folder_contents (structure  & toolbar patterns, https://github.com/plone/mockup/pull/1028), I noticed this issue. Copied from there:

The toolbar updating is done twice. And it's not only the toolbar, for every navigation click in folder_contents, 3 ajax calls are made twice to the backend for:

    @@render-toolbar (toolbar itself)
    @@fc-contextInfo (actions in the toolbar)
    @@getVocabulary (folder_contents main table)

First I suspected event handlers being registered and executed twice. Some jquery .on() registrations are not first removed, which causes events to be stacked when the pattern containing the registration is re-activated, but the event is registered on a tag/container outside the part of the dom being replaced, in particular body or window.  For example, https://github.com/plone/mockup/blob/590d229599433bac4f2265ce7dfa7857c408bf17/mockup/patterns/toolbar/pattern.js#L247  . This one is fixed now, but there are some others left in the toolbar pattern.

Fixing the other relevant event handlers to proper unregister/register with a namespace has at least the effect that the ajax requests are now properly grouped, instead of appearing at random in the network tab. 

<img width="1131" alt="chrome xhr request ordering" src="https://user-images.githubusercontent.com/394784
/97780937-27132380-1b88-11eb-90d5-37171f2b8d5b.png">

If an eventhandler could be propagated and executed twice, adding `e.preventDefault();       e.stopPropagation();` as seen in other event handler functions should helpt, but it doesn't. 

Inspect the timing/stacktrace for @@render-toolbar or the other requests doesn't reveal much at first, they are all the same. The only relevant code paths are from our structure pattern, tablerow, navigation, app, and textfilter.   Textfilter was a suspicious for me, as I am not updating it, I'm navigating. Maybe it re-applies filtering to the collection after navigating?

<img width="1131" alt="event handler stacktrace" src="https://user-images.githubusercontent.com/394784/97781035-e4058000-1b88-11eb-92dc-ab782d6ba663.png">

When I comment this line from the setCurrentPath (from the stacktrace above):

https://github.com/plone/mockup/blob/2a7c5707ecebbf601ed13d5b2ab12485870949a8/mockup/patterns/structure/js/views/app.js#L280

The double xhr requests are gone. The event stacktrace for @@render-toolbar is also much shorter/cleaner

<img width="1112" alt="comment clearfilter" src="https://user-images.githubusercontent.com/394784/97781123-af45f880-1b89-11eb-9493-b8cc718735ef.png">

Only new issue now when I disable clearTerm is that the filter is not reset  when navigating, so you can end up very quickly with an empty collection/folder-listing. 
<img width="949" alt="textfilter-survives-navigation" src="https://user-images.githubusercontent.com/394784/97781293-9d188a00-1b8a-11eb-9153-583ce7b6bc4d.png">

So something in clearTerm is triggering the double folder_contents/toolbar refresh. Digging 2 functions deeper:

https://github.com/plone/mockup/blob/2a7c5707ecebbf601ed13d5b2ab12485870949a8/mockup/patterns/structure/js/views/textfilter.js#L76

Re-initiates the pager in the folder_contents that triggers the extra reload sequence.

**Comment by fredvd** [comment]
@thet Possible fix in https://github.com/plone/mockup/pull/1031  

It's not really pretty to pass booleans through the stack to prevent the page filter. I was unsure if I could use the set_input parameter on setTerm in the textFilter which also seems to be such a parameter, but I kept them separate from_now.

**Comment by fredvd** [comment]
@mauritsvanrees Maybe still in time for 5.2.3 if this is a proper solution?

**Comment by fredvd** [comment]
Update on this issue. In Plone 5.2.4 all serious navigation/update problems with the folder_contents 'web app' should have been solved.  What still happens is that the batching/navigation blocks of the folder_contents and the overall 'keep current context in sync' events (toolbar add content menu, breadcrumbs, upload widget) are entangled .   

So when you go to folder contents there is still one double @@getVocabulary  request, but other xhr requests are done only once.
